### DISCUSSION TITLE
Help Request: Automating User Authentication for Task Pool

### DISCUSSION THREAD

**Discussion by hitaspdotnet** [ulixee/hero#335] [Q&A]
Hi everyone,

I need to fetch data from an endpoint that requires an authenticated user (x.com, graphQL). I’m working with a task pool, handling tasks in parallel, and using a timeout to close the Hero instance if there are no more tasks.

I’m curious to know if there’s a way to create a plugin that performs the login process for every new instance created. This way, I wouldn’t have to worry about authenticating the user on the client side.

Any guidance or examples would be greatly appreciated!

Thank you!

**Comment by blakebyrnes** [comment]
Could a [user profile](https://ulixee.org/docs/hero/advanced-client/user-profile) be useful? It seems like if you can re-use the authentication, it's gotta be in some form of token.
